File: src/main/java/sprykerFileCreator/model/definition/spryker/ClientPluginDefinition.java
package pav.sprykerFileCreator.model.definition.spryker;

import pav.sprykerFileCreator.model.definition.ClassDefinitionInterface;
import pav.sprykerFileCreator.model.generator.SprykerConstants;

public class ClientPluginDefinition implements ClassDefinitionInterface {

    @Override
    public String getClassType() {
        return SprykerConstants.CLIENT_PLUGIN;
    }

    @Override
    public String getNamePattern() {
        return "{{ className }}Plugin";
    }

    @Override
    public String getNamespacePattern() {
        return "{{ projectName }}\\Client\\{{ bundleName }}\\Plugin";
    }

    @Override
    public String getMethodForReturnType() {
        return "";
    }

    @Override
    public String[] getDocBlockClasses() {
        String [] docBlockClasses = {
                SprykerConstants.CLIENT,
                SprykerConstants.CLIENT_FACTORY,
        };

        return docBlockClasses;
    }

    @Override
    public String getDefaultParentClass() {
        return "Spryker\\Client\\Kernel\\AbstractPlugin";
    }

}

File: src/main/kotlin/com/github/tobi812/sprykerplugin/models/definitions/spryker/ClientPluginDefinition.kt
package com.github.tobi812.sprykerplugin.models.definitions.spryker

import com.github.tobi812.sprykerplugin.constants.SprykerConstants
import com.github.tobi812.sprykerplugin.models.definitions.ClassDefinitionInterface

class ClientPluginDefinition : ClassDefinitionInterface {
    override val classType: String
        get() = SprykerConstants.CLIENT_PLUGIN
    override val namePattern: String
        get() = "{{ className }}Plugin"
    override val namespacePattern: String
        get() = "{{ projectName }}\\Client\\{{ bundleName }}\\Plugin"
    override val methodForReturnType: String
        get() = ""
    override val docBlockClasses: Array<String>
        get() = arrayOf(
                SprykerConstants.CLIENT,
                SprykerConstants.CLIENT_FACTORY)
    override val defaultParentClass: String
        get() = "Spryker\\Client\\Kernel\\AbstractPlugin"
}